Reword FileSystem documentation

pachun opened the pull request:
The documentation for `FileSystem.uploadAsync` looks like it may have
been copied from the `FileSystem.downloadAsync` documentation and a
small update that should have been made was overlooked. The word
"download" appears several times in the `uploadAsync` method's
description. This commit rewords those statements to say:

> ... the upload network request ...

rather than

> ... the download network request ...

pachun committed:
Reword “download” to “upload” in FileSystem docs

The documentation for `FileSystem.uploadAsync` looks like it may have
been copied from the `FileSystem.downloadAsync` documentation and a
small update that should have been made was overlooked. The word
"download" appears several times in the `uploadAsync` method's
description. This commit rewords those statements to say:

> ... the upload network request ...

rather than

> ... the download network request ...

barthap commented on docs/pages/versions/v38.0.0/sdk/filesystem.md:
```diff
@@ -363,9 +363,9 @@ Upload the contents of the file pointed by `fileUri` to the remote url.
 Returns a Promise that resolves to an object with the following fields:
 
-- **status (_number_)** -- The HTTP status code for the download network request.
+- **status (_number_)** -- The HTTP status code for the upload network request.
```
```suggestion
- **status (_number_)** -- The HTTP response status code for the upload network request.
```

barthap commented on docs/pages/versions/v38.0.0/sdk/filesystem.md:
```diff
@@ -363,9 +363,9 @@ Upload the contents of the file pointed by `fileUri` to the remote url.
+- **status (_number_)** -- The HTTP status code for the upload network request.
 
-- **headers (_object_)** -- An object containing all the HTTP header fields and their values for the download network request. The keys and values of the object are the header names and values respectively.
+- **headers (_object_)** -- An object containing all the HTTP header fields and their values for the upload network request. The keys and values of the object are the header names and values respectively.
```
```suggestion
- **headers (_object_)** -- An object containing all the HTTP response header fields and their values for the upload network request. The keys and values of the object are the header names and values respectively.
```

barthap reviewed (commented):
Thank you for contribution! 🎉 
I've noticed that both `downloadAsync` and `uploadAsync` return values description refers to _"request"_, but of course it should be _"response"_ 😉  I marked some places, but it's not all of them.

pachun committed:
Specify FileSystem return values

The return values of both the `FileSystem.uploadAsync` and
`FileSystem.downloadAsync` methods state that they return an HTTP status
and HTTP headers. This commit updates those descriptions to specify that
the HTTP status and headers the documentation describes are attributes
of the response received from the server; not the request sent to the
server.

pachun commented:
@barthap Thanks for your feedback.

I included your suggested changes in a separate commit. Happy to squash those before merging. I'm not sure which you prefer.

I also rebased the branch back onto the current version of master, hence the force push.

Let me know what you think and thanks for the help!

barthap reviewed (approved):
Hi, you don't have to squash commits, they're gonna be squashed automatically during merge into one commit named the same as the PR title.

pachun commented:
Ok, good to know!

I updated the PR title to be less specific to the first commit, and more describe the whole of the changes.

pachun commented:
@barthap Just checking in to make sure nobody is waiting for me to do something else in order to merge this. If not, great! If so, happy to do it! Just let me know what needs to be done!

barthap commented:
There's nothing else to do here now. We need to just wait for someone else approving your PR and merging it.

Thanks again for your contribution, I'll be glad to see your further PRs!  🎉

barthap commented:
Hello again @pachun

I'm sorry nobody else reacted to your PR for so long - everybody is consumed with SDK 39 release process.

Now there is another page where your change needs to be done - in `docs/pages/versins/39.0.0`. The simplest way is probably to rebase your branch on current master.

pachun committed:
Reword filesystem docs for SDK 39

1. Reword “download” to “upload” in FileSystem docs

The documentation for `FileSystem.uploadAsync` looks like it may have
been copied from the `FileSystem.downloadAsync` documentation and a
small update that should have been made was overlooked. The word
"download" appears several times in the `uploadAsync` method's
description. This commit rewords those statements to say:

> ... the upload network request ...

rather than

> ... the download network request ...


2. Specify FileSystem return values

The return values of both the `FileSystem.uploadAsync` and
`FileSystem.downloadAsync` methods state that they return an HTTP status
and HTTP headers. This commit updates those descriptions to specify that
the HTTP status and headers the documentation describes are attributes
of the response received from the server; not the request sent to the
server.

pachun commented:
@barthap I rebase on the current master and made the requested changes. Double check me and make sure I didn't goof anything, please!

I have been discouraged that this hasn't been merged yet, but given the understanding is a preoccupation with deploying SDK 39, I want to say "carry on".

SDK updates are little Christmases throughout the year in my world.

Thanks for the feedback and direction!

barthap commented:
Looks good to me 🎉  I'll take care of this to be merged soon.

Tip for you (if you haven't known this already) - you can cd into `docs/` directory, run `yarn && yarn dev` to start documentation server locally. Then you can view your changes live by opening `http://localhost:3000` in your browser. More info [here](https://github.com/expo/expo/blob/master/CONTRIBUTING.md#-updating-documentation). Hope it helps in future contributions!

pachun commented:
@barthap Thank's so much! This is pretty exciting! My name's on Expo!

Thanks again!
